dali/pipeline/data/tensor_list.cc

```diff
@@ -821,6 +852,13 @@ void TensorList<Backend>::Copy(const TensorList<SrcBackend> &src, AccessOrder or
                                bool use_copy_kernel) {
   auto copy_order = copy_impl::SyncBefore(this->order(), src.order(), order);
 
+  if (!src.has_data() && !IsValidType(src.type())) {
+    Reset();
+    SetLayout(src.GetLayout());
+    // no copying to do
```

Comment:
Why not just `if(!src.has_data())`?

Reply:
We can have no allocation but still have a valid type, in that case reset doesn't work that well.

Reply:
So if the source has a valid type, I want to copy it, I leave it to the Resize(<empty_shape>, type) to handle.